File: src/062. Cubic permutations/062.rb
class Integer
  def to_sorted
    self.to_s.chars.sort.join
  end
end

counter = Hash.new {|hash, key| hash[key] = []}
n = 1
result = while true
  cube = n ** 3
  sorted = cube.to_sorted
  arr = counter[sorted] 
  arr << cube
  break arr if arr.size == 5
  n += 1
end

puts arr.min

File: src/062. Cubic permutations/062.java
import java.util.Arrays;
import java.util.Collections;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

public class Main {

    public static void main(String[] args) {
        Map<String, Set<Long>> counter = new HashMap();
        long n = 1;
        while (true) {
            long cube = n * n * n;
            String sorted = toSorted(cube);
            Set<Long> counted = counter.getOrDefault(sorted, new HashSet<>());
            counted.add(cube);
            if (counted.size() == 5) {
                System.out.println(Collections.min(counted));
                break;
            }
            counter.put(sorted, counted);
            n++;
        }
    }

    private static String toSorted(long n) {
        String s = Long.toString(n);
        char[] chars = s.toCharArray();
        Arrays.sort(chars);
        return new String(chars);
    }
}
